tests/saw/spec/handshake/s2n_handshake_io.cry

```diff
@@ -50,15 +52,18 @@ advance_message conn = conn2
                 ,server_can_send_ocsp = conn.server_can_send_ocsp
                 ,key_exchange_eph = conn.key_exchange_eph
                 ,client_auth_flag = conn.client_auth_flag
+                ,actual_protocol_version = conn.actual_protocol_version
                 }
-        (handshake2 : handshake) = {handshake_type = conn.handshake.handshake_type
-                                   ,message_number = conn.handshake.message_number + 1
-                                   }
-        cork2 = if (ACTIVE_STATE conn2).writer != (PREVIOUS_STATE conn2).writer /\
+        (handshake2 : handshake) = { handshake_type = conn.handshake.handshake_type,
+                                     message_number = message_number2 }
+        cork2 = if (ACTIVE_STATE conn2).writer != (ACTIVE_STATE conn).writer /\
                    (ACTIVE_STATE conn2).writer != 'A'
                 then if (ACTIVE_STATE conn2).writer == mode_writer conn2.mode
                      then s2n_cork conn else s2n_uncork conn
                 else conn.corked
+        message_number2 = conn.handshake.message_number + ( if skip_message then 2 else 1)
```

Comment:
Shouldn't this assignment to  `message_number2` be before line 57-58?
```
  (handshake2 : handshake) = { handshake_type = conn.handshake.handshake_type,
                                   ,message_number = conn.handshake.message_number + 1
                                     message_number = message_number2 
```

Reply:
Functional programming: order doesn't matter, because nothing changes state :) Don't think of it as an assignment, think of it as a definition. Since a definition never changes, it can appear anywhere in the file.